amethyst_network/src/simulation/timing.rs

```diff
@@ -0,0 +1,178 @@
+//! Systems and resources to have a consistent, separate simulation frame rate from the ECS
+//! frame rate.
+
+use amethyst_core::{
+    ecs::{Read, System, Write},
+    timing::Time,
+};
+use std::{ops::RangeInclusive, time::Duration};
+
+/// Default number of network simulation frames per second.
+const DEFAULT_SIM_FRAME_RATE: u32 = 30;
+
+/// This system is used exclusively to update the state of the `NetworkSimulationTime` resource.
+pub struct NetworkSimulationTimeSystem;
+
+impl<'s> System<'s> for NetworkSimulationTimeSystem {
+    type SystemData = (Write<'s, NetworkSimulationTime>, Read<'s, Time>);
+
+    fn run(&mut self, (mut sim_time, game_time): Self::SystemData) {
+        sim_time.update_elapsed(game_time.delta_time());
+        sim_time.reset_frame_lag();
+        while sim_time.elapsed_duration() > sim_time.per_frame_duration() {
+            sim_time.increment_frame_number();
+        }
+    }
+}
+
+/// Resource to track the state of the network simulation separately from the ECS frame timings
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct NetworkSimulationTime {
+    /// The current simulation frame
+    frame_number: u32,
+    /// Accumulated duration since last simulation frame
+    elapsed_duration: Duration,
+    /// Duration per frame
+    per_frame_duration: Duration,
+    /// Determines how often we send messages. i.e. "Every N frames" where N is message_send_rate
+    message_send_rate: u8,
+    /// Number of frames behind the simulation is. This will usually be 0 or 1 if the ECS system
```

Comment:
```suggestion
    /// Number of frames behind the simulation is. This will usually be 0 or 1 if the ECS system
```
```suggestion
    /// The number of frames the simulation is behind. This will usually be 0 or 1 if the ECS system
```